service/round_test.go

```diff
@@ -87,10 +88,11 @@ func TestRound_Recovery(t *testing.T) {
 
 	stop()
 	req.ErrorIs(r2recovery1.recoverExecution(ctx, state.Execution, time.Now().Add(duration)), prover.ErrShutdownRequested)
-	require.NoError(t, r2recovery1.waitTeardown(context.Background()))
+	req.NoError(r2recovery1.teardown(false))
 
 	// Recover r2 execution again, and let it complete.
 	ctx, stop = context.WithCancel(context.Background())
+	defer stop()
```

Comment:
👍 there are a few more `context.WithCancel(context.Background())` calls in this test. Can we add a `defer stop()` to those too? (In case a test is terminated early due to a failing assert?)